lib/Sema/TypeCheckCodeCompletion.cpp

```diff
@@ -1325,3 +1349,144 @@ void KeyPathTypeCheckCompletionCallback::sawSolution(
     Results.push_back({BaseType, /*OnRoot=*/(ComponentIndex == 0)});
   }
 }
+
+static bool isInPatternMatch(CodeCompletionExpr *CE, ConstraintSystem &CS) {
+  if (auto *TE = dyn_cast_or_null<TupleExpr>(CS.getParentExpr(CE))) {
+    if (!TE->isImplicit() || TE->getNumElements() != 2 ||
+        TE->getElement(0) != CE)
+      return false;
+    if (auto *DRE = dyn_cast<DeclRefExpr>(TE->getElement(1))) {
+      auto Name =  DRE->getDecl()->getName();
+      return Name && Name.isSimpleName("$match");
+    }
+  }
+  return false;
+}
+
+void ArgumentTypeCheckCompletionCallback::sawSolution(
+    const constraints::Solution &S) {
+  TypeCheckCompletionCallback::sawSolution(S);
+
+  Type ExpectedTy = getTypeForCompletion(S, CompletionExpr);
+  if (!ExpectedTy)
+    return;
+
+  auto &CS = S.getConstraintSystem();
+
+  Expr *ParentCall = CompletionExpr;
+  while (ParentCall &&
+         !(isa<CallExpr>(ParentCall) || isa<SubscriptExpr>(ParentCall) || isa<BinaryExpr>(ParentCall)))
```

Comment:
Should this also be checking `ParentCall->getArgs()` for null instead of matching parent types?